### PR TITLE
Document some more Querier command-line arguments

### PR DESCRIPTION
**Pull request by bboreham**
Triggered by me realising I didn't know whether I wanted `-querier.batch-iterators` or `-querier.iterators`

### CODE REVIEW

File: docs/arguments.md [hunk 1]
```diff
@@ -1,7 +1,57 @@
...
+
+   Address of query frontend service, used by workers to find the frontend which will give them queries to execute.
+
+- `-querier.dns-lookup-period`
+
+   How often the workers will query DNS to re-check where the frontend is.
+
+- `-querier.worker-parallelism`
+
+   Number of simultaneous queries to process, per worker process.
```
Comment (tomwilkie) [inline docs/arguments.md]: These 3 need to be specified on the querier, not the query-frontend.

Comment (bboreham) [reply docs/arguments.md]: moved

**Comment by tomwilkie** [comment]
Sorry was half way through review and got distracted.  More incoming!

File: docs/arguments.md [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
 
+## Querier
+
+- `-querier.max-concurrent`
+
+   The maximum number of top-level PromQL queries that will execute at the same time, per querier process.
+
+- `-querier.query-parallelism`
+
+   This refers to database queries against the store (e.g. Bigtable or DynamoDB).  This is the max subqueries run in parallel per higher-level query.
```
Comment (tomwilkie) [inline docs/arguments.md]: Ah, the flag description is misleading.  "Sub-query" means promql query, but its subquery in that a single promql query is sharded into multiple single day queries.

This should be set to `max-concurrent / # query-frontend replicas`, as this is the number of goroutines which connect to each frontend in parallel.  If you have `query-parallelism > (max-concurrent / # query-frontend replicas)`, then you'll have queries queueing in the queriers and not the frontend, which will affect QoS.

Has no effect if you're not running the query-frontend.

File: docs/arguments.md [hunk 3]
```diff
@@ -1,5 +1,57 @@
...
+
+- `-querier.timeout`
+
+   The timeout for a top-level PromQL query.
+
+- `-querier.max-samples`
+
+   Maximum number of samples a single query can load into memory, to avoid blowing up on enormous queries.
```
Comment (tomwilkie) [inline docs/arguments.md]: Where is this flag defined?   I don't recall it, or any limit like it in the querier.

File: docs/arguments.md (hunk 2 above)
Comment (tomwilkie) [reply docs/arguments.md]: I'm wrong here...

File: docs/arguments.md [hunk 4]
```diff
@@ -1,5 +1,57 @@
...
+
+   Address of query frontend service, used by workers to find the frontend which will give them queries to execute.
+
+- `-querier.dns-lookup-period`
+
+   How often the workers will query DNS to re-check where the frontend is.
+
+- `-querier.worker-parallelism`
+
+   Number of simultaneous queries to process, per worker process.
```
Comment (tomwilkie) [inline docs/arguments.md]: This should be set to `max-concurrent / # query-frontend` replicas, as this is the number of goroutines which connect to each frontend in parallel. If you have `query-parallelism > (max-concurrent / # query-frontend replicas)`, then you'll have queries queueing in the queriers and not the frontend, which will affect QoS.

File: docs/arguments.md [hunk 5]
```diff
@@ -1,5 +1,57 @@
...
+
+   This uses iterators to execute query, as opposed to fully materialising the series in memory, and fetches multiple results per loop.
+
+- `-querier.ingester-streaming`
+
+   Use streaming RPCs to query ingester, to reduce memory pressure in the ingester.
+
+- `-querier.iterators`
+
+   This is similar to `-querier.batch-iterators` but less efficient.
```
Comment (tomwilkie) [inline docs/arguments.md]: Don't set this and `batch-iterators` both to true, as `iterators` will take precedent.  And `batch-iterators` is what most people should use.

File: docs/arguments.md [hunk 6]
```diff
@@ -1,5 +1,57 @@
...
+
+- `-promql.lookback-delta`
+
+   Time since the last sample after which a time series is considered stale and ignored by expression evaluations.
```
Comment (tomwilkie) [inline docs/arguments.md]: This is a prometheus flag - should we document it here?

**Comment by tomwilkie** [comment]
Review done!

File: docs/arguments.md (hunk 3 above)
Comment (bboreham) [reply docs/arguments.md]: https://github.com/cortexproject/cortex/blob/ed1292b5b208e045e881d5f8507fc60ce9e3ee21/pkg/querier/querier.go#L43

It's a Prometheus limit really.

File: docs/arguments.md (hunk 6 above)
Comment (bboreham) [reply docs/arguments.md]: I don't see why not.
https://github.com/cortexproject/cortex/blob/ed1292b5b208e045e881d5f8507fc60ce9e3ee21/pkg/querier/querier.go#L38

File: docs/arguments.md (hunk 4 above)
Comment (bboreham) [reply docs/arguments.md]: I think I might express that as a constraint on `max-concurrent`

**Comment by bboreham** [comment]
@tomwilkie thanks for review; PTAL at latest.

**Comment by tomwilkie** [comment]
LGTM!

**Pull request merged**